File: Solo/Puzzles/Easy/horse-racing-duals/HorseRacingDuals.php
<?php
/**
 * Optimized solution with debugging
 **/

// Lire le nombre de chevaux
fscanf(STDIN, "%d", $N);

// Lire les puissances des chevaux dans un tableau
$horses = [];
for ($i = 0; $i < $N; $i++) {
    fscanf(STDIN, "%d", $horses[]);
}

// Trier les puissances
sort($horses);

// Validation : Vérifiez si le tri est correct (à retirer une fois confirmé)
for ($i = 1; $i < $N; $i++) {
    if ($horses[$i] < $horses[$i - 1]) {
        error_log("Erreur de tri : {$horses[$i - 1]} > {$horses[$i]}");
    }
}

// Initialiser la différence minimale
$minDifference = PHP_INT_MAX;

// Calculer la différence minimale entre les puissances consécutives
for ($i = 0; $i < $N - 1; $i++) {
    $diff = $horses[$i + 1] - $horses[$i];
    if ($diff < $minDifference) {
        $minDifference = $diff;
    }
}

// Afficher la différence minimale
echo $minDifference . "\n";
?>

File: Solo/Puzzles/Easy/horse-racing-duals/HorseRacingDuals.java
import java.util.*;

class Solution {
    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        
        // Lire le nombre de chevaux
        int N = in.nextInt();
        
        // Stocker les puissances des chevaux dans un tableau
        int[] horses = new int[N];
        for (int i = 0; i < N; i++) {
            horses[i] = in.nextInt();
        }
        
        // Trier les puissances
        Arrays.sort(horses);
        
        // Calculer la différence minimale
        int minDifference = Integer.MAX_VALUE;
        for (int i = 1; i < N; i++) {
            int diff = horses[i] - horses[i - 1];
            if (diff < minDifference) {
                minDifference = diff;
            }
        }
        
        // Afficher la différence minimale
        System.out.println(minDifference);
    }
}
